### ISSUE TITLE
[CI] org.elasticsearch.bwcompat.BasicAnalysisBackwardCompatibilityIT.testAnalyzerTokensAfterUpgrade fails on 2.4 branch

### ISSUE THREAD

**Issue by dakrone** [elastic/elasticsearch#24345]
It fails with the Hindi analyzer:

```
java.lang.AssertionError: failed for term: 4𐒧 field: field_144 analyzer: type=string,analyzer=hindi input: hstyso ukpr 5\uf5bd\u011e vz \ud801\udca4\ud801\udca7 \u1cd2\u1cd9 uoeupwfvx  ks kyqgzqiojod \u31fc\u31f9 \u0766 jcuot )c(j
Expected: "47"
     but: was "4\ud801\udca7"
	at __randomizedtesting.SeedInfo.seed([2C42684E19A0B262:228231FD5F0AF676]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:865)
	at org.elasticsearch.bwcompat.BasicAnalysisBackwardCompatibilityIT.testAnalyzerTokensAfterUpgrade(BasicAnalysisBackwardCompatibilityIT.java:100)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1764)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:871)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:907)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:921)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
```

And reproduces consistently for me with:

```
mvn verify -Pdev -Dskip.unit.tests -pl org.elasticsearch.qa.backwards:2.3 -Dtests.seed=2C42684E19A0B262 -Dtests.class=org.elasticsearch.bwcompat.BasicAnalysisBackwardCompatibilityIT -Dtests.method="testAnalyzerTokensAfterUpgrade" -Des.logger.level=ERROR -Dtests.assertion.disabled=false -Dtests.security.manager=true -Dtests.heap.size=512m -Dtests.locale=he -Dtests.timezone=MST7MDT
```

See: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+2.4+multijob-intake/105/consoleText

I'm not sure what's recently changed in 2.3 or 2.4 to cause this, but it looks like this test was re-enabled in 2014 by @rjernst, so do you know who'd be a good person to take a look @rjernst?

**Comment by rjernst** [comment]
The test was re-enabled after it broke between lucene 4.9 and 4.10, because of a behavior change in refactoring of standard tokenizer.  When the test was re-enabled, I added a loop to ensure we don't generate input that could hit the behavior difference.

This failure is different. Since lucene 5.4, we have done ascii number folding. So any digits in other scripts are replaced by their ascii equivalent. In this test, there are 2 surrogate pairs of numbers, which should be folded to `47`. But the second pair is left unfolded. Looking at the code in `DecimalDigitFilter`, it seems this could only happen if java did not think this character was a number.  I tried these on master and they work fine.  Maybe @jpountz has another idea on what could be wrong?  @dakrone which version of java are you running?
